app/controllers/UserController.py

```diff
@@ -0,0 +1,196 @@
+from app.models.user import User
+from app import response, app, db
+from app.blacklist import blacklist
+from flask import request, jsonify
+import datetime
+from flask_jwt_extended import *
+
+def getAllUsers():
+    try:
+        users = User.query.all()
+        return jsonify({'users': [
+            {
+                'id': user.id,
+                'name': user.name,
+                'email': user.email,
+                'school': user.school,
+                'role': user.role,
+                'grade': user.grade,
+                'dob': user.dob,
+                'created_at': user.created_at,
+                'updated_at': user.updated_at
+            } for user in users
+        ]})
+    except Exception as e:
+        print(e)
+
+def getUserByID(id):
+    try:
+        user = User.query.get(id)
+
+        if user:
+            data = {
+                'id': user.id,
+                'name': user.name,
+                'email': user.email,
+                'school': user.school,
+                'role': user.role,
+                'grade': user.grade,
+                'dob': user.dob,
+                'created_at': user.created_at,
+                'updated_at': user.updated_at
+            }
+            return jsonify(data)
+        else:
+            return jsonify({'error': 'User not found'}), 404
+        
+    except Exception as e:
+        print(e)
+
+def register():
+    try:
+        name = request.form.get('name')
+        email = request.form.get('email')
+        password = request.form.get('password')
+        school = request.form.get('school')
+        role = request.form.get('role')
+        grade = request.form.get('grade')
+        dob = 0
+
+        user = User(
+            name=name,
+            email=email,
+            school=school,
+            role = role,
+            grade=grade,
+            dob=dob
+            )
+        user.setPassword(password)
+        db.session.add(user)
+        db.session.commit()
+
+        return response.success('', 'User has been created')
+    except Exception as e:
+            print(e)
+
+def updateUser(id):
```

Comment:
the function name should be `update_user`

Reply:
done